### ISSUE TITLE
Paramètres par défaut de la table "layer_styles"

### ISSUE THREAD

**Issue by StormKlaus** [MTES-MCT/asgard-manager#8]

**Comment by alhyss** [comment]
@StormKlaus J'ai transféré l'issue dans le Git d'AsgardManager, puisque c'est essentiellement ce module qui pourrait être concerné.

Je ne suis preneuse d'un contre-argumentaire, mais, en première approche, je ne pense pas qu'il soit vraiment judicieux d'avoir dans AsgardManager une fonctionnalité qui altère la structure d'une table de QGIS, sans que nous ne maîtrisions les effets de bord potentiels (est-ce que ce nombre maximum de caractères pourrait avoir une raison d'être ?), ni la manière dont la table va évoluer avec les versions. Il me semble préférable d'expliquer [dans la documentation](https://snum.scenari-community.org/Asgard/Documentation/co/SEC_QuidSchemaPublic.html) que ce problème peut survenir et donner la commande qui permet de passer outre la limite, que l'ADL choisira d'exécuter à ses risques et périls...

**Comment by StormKlaus** [comment]
Merci pour ce retour.

Pas de contre-argumentaire, suite à tes explications je te rejoins.

Si je comprend bien ce serait plus une évolution à faire remonter sur le Gihub de QGIS
(https://github.com/qgis/QGIS/issues)

Pour les noms d'utilisateurs > à 30, on pourrait peut-être envisager une contrainte/limite dans Asgard Manager lors de la création des rôles de connexions, ou un message d'avertissement...
Mais je sais pas finalement s'il est vraiment utile de développer quoi que ce soit à ce propos...

Évoquer ce sujet dans la documentation serait un plus.

Merci

**Comment by alhyss** [comment]
Oui, je pense que c'est une bonne idée de remonter la question côté QGIS, si ça n'a pas déjà été fait. Si la limitation de longueur ne sert à rien, il serait beaucoup plus propre qu'ils corrigent ça eux-mêmes.

Pour les noms de rôle, je ne sais pas trop comment on pourrait expliquer aux utilisateurs d'où vient cette limite à 30 caractères, qui n'a d'ailleurs un sens que s'ils se servent de `layer_styles`... @WREATCHED Que fait AsgardManager aujourd'hui pour les noms de plus de 63 caractères (qui sont tronqués par PostgreSQL) ? On pourrait imaginer une longueur limite paramétrable avec une valeur par défaut à 63 caractères ? En conseillant dans la doc de l'abaisser à 30 pour les usagers de `layer_styles` ?

**Comment by StormKlaus** [comment]
Effectivement, le problème a déjà été remonté sur le Github de QGIS, plusieurs issues à ce propos :
https://github.com/qgis/QGIS/issues/47550
https://github.com/qgis/QGIS/issues/39850
https://github.com/qgis/QGIS/blob/8deed6c49742bf01d4d49a369fe52a7100fcf107/src/providers/postgres/qgspostgresprovider.cpp

Et donc si j'ai bien compris ce problème a bien été résolu dans les versions > ou alentours à la QGIS 3.0.
Il n'y aurait désormais plus de limite de caractère pour les noms de couches.

Le problème est que la table layer_syles a été créé avec une version QGIS < à la 3.0 chez nous et nous sommes (étions) donc toujours contraint par cette limite.

Donc pour être en cohérence avec les dernières versions de QGIS la requête à utiliser serait plutôt celle-ci : 
ALTER TABLE public.layer_styles ALTER COLUMN stylename TYPE text ;

**Comment by alhyss** [comment]
Merci d'avoir cherché !

Si le problème est déjà résolu pour les versions de QGIS prises en charge par AsgardManager, alors il n'y a vraiment pas lieu de prévoir un correctif de notre côté. Une mention dans la doc devrait être suffisante. `owner` est limité à 63 caractères (et l'est toujours dans la [version courante](https://github.com/qgis/QGIS/blob/baea1e89278ea40ba16f113d38ee1cdafa83bb3c/src/providers/postgres/qgspostgresprovider.cpp#L5438)), mais c'est la limite de PostgreSQL de toute façon, donc il ne devrait pas y avoir de problème de ce côté pour les nouvelles tables `layer_styles`.
